### PR TITLE
16 registration service

### LINKED ISSUE #16: Registration service
Create endpoint for register new user in the **backend** side.

Write services for registration. And add all necessary dependency

### PR DESCRIPTION
**Pull request by maryanchuk04**
This PR contain full **registration** new User account.

I create some:
-  services
- api controller
- migrations

### CODE REVIEW

**Review by Vova-Romaniuk — APPROVED** [review]

File: StudCity.Backend/StudCity.API/Program.cs [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+
+#endregion
 #region ConfigureServices
-Console.WriteLine(builder.Configuration.GetConnectionString("ApplicationDbConnectionString"));
+builder.Services.AddSingleton(jwtConfiguration);
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddDbContextFactory<StudCityContext>(
     options => options.UseSqlServer(builder.Configuration.GetConnectionString("ApplicationDbConnectionString"),
         b => b.MigrationsAssembly("StudCity.Db")), ServiceLifetime.Scoped);
```
Comment (Serhiy-dotcom) [inline StudCity.Backend/StudCity.API/Program.cs]: Please add additional paragraph for third parameter

File: StudCity.Backend/StudCity.Application/Helpers/PasswordHasherService.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace StudCity.Application.Helpers;
+
+public class PasswordHasherService : IPasswordHasher
+{
+    public string HashPassword(string password) => BCrypt.Net.BCrypt.HashPassword(password);
+    
+    public bool VerifyPassword(string password, string hashingString)
+    {
+        return BCrypt.Net.BCrypt.Verify(password,hashingString);
```
Comment (Serhiy-dotcom) [inline StudCity.Backend/StudCity.Application/Helpers/PasswordHasherService.cs]: add a space before 'hashingString' prop

File: StudCity.Backend/StudCity.Application/Providers/TokenProvider.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        return new AccountToken()
+        {
+            Token = _pinGenerator.Generate().ToString(),
+            Expires = DateTime.Now.AddDays(7),
+            Created = DateTime.Now,
+            AccountId = accountId,
+            Type = TokenType.EmailConfirmationType
+        };
+
```
Comment (Serhiy-dotcom) [inline StudCity.Backend/StudCity.Application/Providers/TokenProvider.cs]: remove paragraph

**Review by Serhiy-dotcom — CHANGES_REQUESTED** [review]
Please fix the issues below.

**Review by Serhiy-dotcom — APPROVED** [review]
Very good!

**Pull request merged**